File: netlify/functions/get_calendar.py
import json
import os
import psycopg2

def handler(event, context):
    try:
        conn = psycopg2.connect(os.environ["DATABASE_URL"])
        cur = conn.cursor()
        cur.execute("SELECT upload_date, status FROM uploads")
        rows = cur.fetchall()
        cur.close()
        conn.close()
        result = {str(r[0]): r[1] for r in rows}
        return {"statusCode": 200, "body": json.dumps(result)}
    except Exception as e:
        return {"statusCode": 500, "body": json.dumps({"error": str(e)})}

File: netlify/functions/get_calendar.mjs
import { getDatabase } from "@netlify/database";

export default async () => {
  try {
    const db = getDatabase();
    const rows = await db.sql`SELECT upload_date, status FROM uploads`;
    const result = {};
    for (const r of rows) {
      const d = r.upload_date instanceof Date
        ? r.upload_date.toISOString().slice(0, 10)
        : String(r.upload_date).slice(0, 10);
      result[d] = r.status;
    }
    return Response.json(result);
  } catch (e) {
    return Response.json({ error: e.message }, { status: 500 });
  }
};

export const config = {
  path: "/.netlify/functions/get_calendar",
};
